pytest :test_load_structure fails

Hi!
I have installed pyseer through cloning the git repository and now am running a pytest. However, the test_load_structure keeps on failing.
Here are my error messages:

tests/input_test.py::TestLoadFunctions::test_load_structure FAILED  
----
tests/input_test.py:77
----
E           TypeError: __init__() takes from 1 to 2 positional arguments but 3 positional arguments (and 3 keyword-only arguments) were given

pyseer/input.py:116: TypeError

What could have gone wrong?

--- a/pyseer/__main__.py
+++ b/pyseer/__main__.py
@@ -20,7 +20,6 @@ with set_env(MKL_NUM_THREADS='1',
 from scipy.stats import norm
 import scipy.sparse
 import pandas as pd
-from sklearn import manifold
 from multiprocessing import Pool
 
 from .__init__ import __version__

--- a/pyseer/input.py
+++ b/pyseer/input.py
@@ -113,7 +113,7 @@ def load_structure(infile, p, max_dimensions, mds_type="classic", n_cpus=1,
         elif mds_type != "metric":
             sys.stderr.write("Unsupported mds type chosen. Assuming metric\n")
 
-        mds = manifold.MDS(max_dimensions, metric_mds, n_jobs=n_cpus,
+        mds = manifold.MDS(n_components=max_dimensions, metric=metric_mds, n_jobs=n_cpus,
                            dissimilarity='precomputed',
                            random_state=seed)
         projection = mds.fit_transform(m.values)

--- a/tests/compare_tests
+++ b/tests/compare_tests
@@ -38,7 +38,7 @@ if __name__ == "__main__":
             sys.stderr.write('Different outputs shape\n')
             sys.exit(1)
 
-        idx = set(m1.index).intersection(m2.index)
+        idx = list(set(m1.index).intersection(m2.index))
         if len(idx) != m1.shape[0] or len(idx) != m2.shape[0]:
             sys.stderr.write('Different variants\n')
             sys.exit(1)

